spatialos-sdk/src/worker/schema.rs

```diff
@@ -0,0 +1,582 @@
+                unsafe {
+                    data.set_len(count);
+                    $schema_get_list(object.internal, field, data.as_mut_ptr());
+                }
+            }
+        }
+    };
+}
+
+impl_primitive_field!(
+    f32,
+    SchemaFloat,
+    Schema_GetFloat,
+    Schema_IndexFloat,
+    Schema_GetFloatCount,
+    Schema_AddFloat,
+    Schema_AddFloatList,
+    Schema_GetFloatList,
+);
+impl_primitive_field!(
+    f64,
+    SchemaDouble,
+    Schema_GetDouble,
+    Schema_IndexDouble,
+    Schema_GetDoubleCount,
+    Schema_AddDouble,
+    Schema_AddDoubleList,
+    Schema_GetDoubleList,
+);
+impl_primitive_field!(
+    i32,
+    SchemaInt32,
+    Schema_GetInt32,
+    Schema_IndexInt32,
+    Schema_GetInt32Count,
+    Schema_AddInt32,
+    Schema_AddInt32List,
+    Schema_GetInt32List,
+);
+impl_primitive_field!(
+    i64,
+    SchemaInt64,
+    Schema_GetInt64,
+    Schema_IndexInt64,
+    Schema_GetInt64Count,
+    Schema_AddInt64,
+    Schema_AddInt64List,
+    Schema_GetInt64List,
+);
+impl_primitive_field!(
+    u32,
+    SchemaUint32,
+    Schema_GetUint32,
+    Schema_IndexUint32,
+    Schema_GetUint32Count,
+    Schema_AddUint32,
+    Schema_AddUint32List,
+    Schema_GetUint32List,
+);
+impl_primitive_field!(
+    u64,
+    SchemaUint64,
+    Schema_GetUint64,
+    Schema_IndexUint64,
+    Schema_GetUint64Count,
+    Schema_AddUint64,
+    Schema_AddUint64List,
+    Schema_GetUint64List,
+);
+impl_primitive_field!(
+    i32,
+    SchemaSint32,
+    Schema_GetSint32,
+    Schema_IndexSint32,
+    Schema_GetSint32Count,
+    Schema_AddSint32,
+    Schema_AddSint32List,
+    Schema_GetSint32List,
+);
+impl_primitive_field!(
+    i64,
+    SchemaSint64,
+    Schema_GetSint64,
+    Schema_IndexSint64,
+    Schema_GetSint64Count,
+    Schema_AddSint64,
+    Schema_AddSint64List,
+    Schema_GetSint64List,
+);
+impl_primitive_field!(
+    u32,
+    SchemaFixed32,
+    Schema_GetFixed32,
+    Schema_IndexFixed32,
+    Schema_GetFixed32Count,
+    Schema_AddFixed32,
+    Schema_AddFixed32List,
+    Schema_GetFixed32List,
+);
+impl_primitive_field!(
+    u64,
+    SchemaFixed64,
+    Schema_GetFixed64,
+    Schema_IndexFixed64,
+    Schema_GetFixed64Count,
+    Schema_AddFixed64,
+    Schema_AddFixed64List,
+    Schema_GetFixed64List,
+);
+impl_primitive_field!(
+    i32,
+    SchemaSfixed32,
+    Schema_GetSfixed32,
+    Schema_IndexSfixed32,
+    Schema_GetSfixed32Count,
+    Schema_AddSfixed32,
+    Schema_AddSfixed32List,
+    Schema_GetSfixed32List,
+);
+impl_primitive_field!(
+    i64,
+    SchemaSfixed64,
+    Schema_GetSfixed64,
+    Schema_IndexSfixed64,
+    Schema_GetSfixed64Count,
+    Schema_AddSfixed64,
+    Schema_AddSfixed64List,
+    Schema_GetSfixed64List,
+);
+impl_primitive_field!(
+    u32,
+    SchemaEnum,
+    Schema_GetEnum,
+    Schema_IndexEnum,
+    Schema_GetEnumCount,
+    Schema_AddEnum,
+    Schema_AddEnumList,
+    Schema_GetEnumList,
+);
+
+impl SchemaType for EntityId {
```

Comment:
So we probably *could* use the macro here, if we added an `Into` conversion between `EntityId` and `i64`. I was hesitant to do that, though, because I'd like to make `EntityId` more strongly-typed (i.e. disallow converting between `i64` and `EntityId`, effectively treat `EntityId` as it's own type), but I don't know if that's a reasonable thing to do. Thoughts?

Reply:
I don't think its _that_ crazy to provide the `Into` conversion. The C lib just has a type alias.